<launch>
    <rosparam command="load" file="$(find project11_vorc)/config/vorc.yaml" />
    
    <arg name="viz" default="false" />
    <arg name="perception" default="true" />
    <arg name="rqt" default="false"/>
    <arg name="camp" default="false" />
    <arg name="simpler" default="true"/>
    <arg name="gui" default="false"/>
    <arg name="operator" default="false"/>

    <include file="$(find project11_vorc)/launch/localization.launch"/>
    
    
    <node pkg="cora_helm" type="cora_helm_node.py" name="cora_helm"/>

    <node unless="$(arg simpler)" pkg="cora_helm" type="simple_differential_controller.py" name="simple_controller">
        <remap from="odom" to="/cora/robot_localization/odometry/filtered"/>
    </node>
    <node if="$(arg simpler)" pkg="cora_helm" type="simpler_differential_controller.py" name="simpler_controller">
        <remap from="odom" to="/cora/robot_localization/odometry/filtered"/>
    </node>

    <node pkg="dp_hover" type="dp_hover_node" name="dp_hover"/>
    
    <include file="$(find project11_vorc)/launch/navigation.launch"/>
    
    <include if="$(arg perception)" file="$(find project11_vorc)/launch/darknet.launch">
        <arg name="gui"     value="$(arg gui)" />
    </include>
    
    <node pkg="project11_vorc" type="vorc_task_manager.py" name="task_manager"/>
    
    <include if="$(arg viz)" file="$(find project11_vorc)/launch/viz.launch"/>
  
    <node if="$(arg rqt)" pkg="rqt_gui" type="rqt_gui" name="rqt"/>

    <node if="$(arg operator)" pkg="joy" type="joy_node" name="joy_node">
        <param name="autorepeat_rate" value="10.0"/>
    </node>
    
    <param name="allow_differential_drive" value="True"/>
    <node if="$(arg operator)" pkg="joy_to_helm" type="joy_to_helm.py" name="joy_to_helm"/>

    <node if="$(arg operator)" pkg="command_bridge" type="command_bridge_sender_node.py" name="command_bridge_sender"/>
    <node if="$(arg operator)" pkg="command_bridge" type="command_bridge_receiver_node.py" name="command_bridge_receiver"/>

    <node if="$(arg camp)" pkg="camp" type="AutonomousMissionPlanner" name="camp"  args="$(find project11_vorc)/charts/ $(find project11_vorc)/charts/L16-23800-34552-16-16_16.kap"/>
    <node if="$(arg camp)" pkg="project11_vorc" type="occupancy_grid_to_camp.py" name="grid_to_camp">
        <remap from="occupancy_grid" to="/move_base/local_costmap/costmap"/>
        <!--<remap from="occupancy_grid" to="/move_base/global_costmap/costmap"/>-->
    </node>

    <node pkg="rosbag" type="record" name="vorc_logging" args="-o /home/developer/.ros/log/
            /vorc/task/info
            /position
            /posmv/orientation
            /posmv/position
            /project11/display
            /project11/mission_manager/status
            /heartbeat
            /heading
            /differential_drive
            /darknet_ros/bounding_boxes
            /darknet_ros/detection_image
            /clock
            /cmd_vel
            /vorc/station_keeping/goal
            /vorc/wayfinding/waypoints
            /vorc/perception/landmark
    " />    

    
</launch>
